src/gas/gas-util.test.ts

```diff
@@ -155,72 +227,741 @@ describe('gas utils', () => {
     });
+  });
 
-    it('should round to whole WEI numbers', () => {
-      expect(normalizeGWEIDecimalNumbers(0.1001)).toBe('0.1001');
-      expect(normalizeGWEIDecimalNumbers(0.0109)).toBe('0.0109');
-      expect(normalizeGWEIDecimalNumbers(0.0014)).toBe('0.0014');
-      expect(normalizeGWEIDecimalNumbers(0.5676)).toBe('0.5676');
-    });
+  describe('calculateTimeEstimate', () => {
+    const buildGasFeeEstimates = ({
+      estimatedBaseFee,
+      low = {},
+      medium = {},
+      high = {},
+    }: {
+      estimatedBaseFee: string;
+      low?: {
+        minWaitTimeEstimate?: number;
+        maxWaitTimeEstimate?: number;
+        suggestedMaxPriorityFeePerGas?: string;
+      };
+      medium?: {
+        minWaitTimeEstimate?: number;
+        maxWaitTimeEstimate?: number;
+        suggestedMaxPriorityFeePerGas?: string;
+      };
+      high?: {
+        minWaitTimeEstimate?: number;
+        maxWaitTimeEstimate?: number;
+        suggestedMaxPriorityFeePerGas?: string;
+      };
+    }): GasFeeEstimates => {
+      return {
+        low: {
+          minWaitTimeEstimate: 0,
+          maxWaitTimeEstimate: 0,
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...low,
+        },
+        medium: {
+          minWaitTimeEstimate: 0,
+          maxWaitTimeEstimate: 0,
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...medium,
+        },
+        high: {
+          minWaitTimeEstimate: 0,
+          maxWaitTimeEstimate: 0,
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...high,
+        },
+        estimatedBaseFee,
+        historicalBaseFeeRange: null,
+        baseFeeTrend: null,
+        latestPriorityFeeRange: null,
+        historicalPriorityFeeRange: null,
+        priorityFeeTrend: null,
+        networkCongestion: null,
+      };
+    };
 
-    it('should handle inputs with more than 9 decimal places', () => {
-      expect(normalizeGWEIDecimalNumbers(1.0000000162)).toBe('1.000000016');
-      expect(normalizeGWEIDecimalNumbers(1.0000000165)).toBe('1.000000017');
-      expect(normalizeGWEIDecimalNumbers(1.0000000199)).toBe('1.00000002');
-      expect(normalizeGWEIDecimalNumbers(1.9999999999)).toBe('2');
-      expect(normalizeGWEIDecimalNumbers(1.0000005998)).toBe('1.0000006');
-      expect(normalizeGWEIDecimalNumbers(123456.0000005998)).toBe(
-        '123456.0000006',
-      );
-      expect(normalizeGWEIDecimalNumbers(1.000000016025)).toBe('1.000000016');
-      expect(normalizeGWEIDecimalNumbers(1.0000000160000028)).toBe(
-        '1.000000016',
-      );
-      expect(normalizeGWEIDecimalNumbers(1.000000016522)).toBe('1.000000017');
-      expect(normalizeGWEIDecimalNumbers(1.000000016800022)).toBe(
-        '1.000000017',
-      );
-    });
+    describe('if the given priority fee is less than the given max fee minus the latest base fee', () => {
```

Comment:
Nit: This was easier for me to understand when I rephrased it to be "If the priority + base fee is (less than/equal to/exceeds) the max fee".  The "A < B - C" was a lot to parse.

I wish there was a better way to explain this than that even :thinking: But I can't think of anything better.

Reply:
Makes sense, I can make this change.